### PR TITLE
[ci] add dependency verification action

### PR DESCRIPTION
**Pull request by metajack**
This adds a new GitHub Actions workflow to report on dependency changes to specific subsets of Libra. It currently tests the safety rules TCB and the execution correctness TCB as well as all the validator release binaries.

It will comment and label PRs that make such changes, as well as notifying @davidiw when TCB deps are changed.

This effectively replaces cargo-dep-bot.

### CODE REVIEW

**Comment by metajack** [comment]
Note that I will update the in-tree sumarries to the latest  before I land this. Otherwise it will trigger on every PR.

Also, I need to add support for removing comments and labels if the PR changes to not have dep changes.

**Comment by metajack** [comment]
Nice, comments and labels were deleted by the action. This is ready to go!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -5,3 +5,6 @@
 /client/libra-dev/ @thefallentree @xli
 # Subscribe to LCS format changes
 /testsuite/generate-format/tests/staged/libra.yaml @thefallentree @matbd @xli
+# Subscribe to TCB dependency summary changes
+/summaries/summary-lsr.toml @davidiw
+/summaries/summary-lec.toml @davidiw
```
Comment (bmwill) [inline .github/CODEOWNERS:10]: Do we want an owner for the release summary?

File: summaries/summary-lec.toml [hunk 2]
```diff
@@ -0,0 +1,1595 @@
+# Summary for Libra subset 'lec'. @generated by x.
+# To regenerate, run 'cargo x generate-summaries'.
```
Comment (bmwill) [inline summaries/summary-lec.toml:2]: Actually thinking about this more, we probably can't merge this yet until we can block landing  if these summary files are out of date. The last thing we want is the action reporting differences when a PR doesn't actually include any (but a previously merged PR into master had differences but didn't update these files)

Comment (bmwill) [reply summaries/summary-lec.toml:2]: So we probably just need to add a step in CI that just runs `cargo x generate-summaries` and ensures that no files are modified after that.

Comment (metajack) [reply summaries/summary-lec.toml:2]: I'll add something to fail the PR if the sumamries don't match.

File: .github/CODEOWNERS (hunk 1 above)
Comment (metajack) [reply .github/CODEOWNERS:10]: I was only going to enforce the TCB with the extra review as a first step. We can enforce the release binaries later if this goes well.

**Comment by metajack** [comment]
Ok, so the desired behavior should be to pass/fail based on summary differences, but post comment/label should always happen with the delta between the merge base and the new calculated summary. I'll fix that up.

**Comment by metajack** [comment]
Ok, new behavior added.

**Pull request closed without merging**